src/_guides/language/evolution.md

```diff
@@ -313,74 +338,74 @@ multiple versions of the Dart language.
+Language versioning becomes important on the rare occasions when Dart
+introduces an incompatible feature like [null safety][].
+When Dart introduces a breaking change, code that did compile
+might no longer compile. Using language versions allows you to indicate
+the compatibility of the code using the `pubspec.yaml` file.
+
+In the case of null safety, Dart versions 2.12 through 2.19 allowed you
+to _choose_ to update your code to use null safety.
+Dart uses language versioning to permit non-null-safe code to run
+alongside null-safe code.
+This decision enabled migration from non-null-safe to null-safe code.
+To check out an example of how an app or package can migrate to a new
+language version with an incompatible feature, check out
 [Migrating to null safety](/null-safety/migration-guide).
 
-Each package has a default language version,
-equal to the `<major>.<minor>` part of the
-**lower SDK constraint** in the pubspec.
-For example, the following entry in a `pubspec.yaml` file
-indicates that this package uses the Dart 2.7 language version.
+Each package has a default language version, equal to the
+`<major>.<minor>` part of the **lower SDK constraint** in the pubspec.
+
+**For example:** the following entry in a `pubspec.yaml` file
+indicates that this package uses the Dart 2.18 language version.
 
 ```yaml
 environment:
-  sdk: ">=2.7.0 <3.0.0"
+  sdk: ^2.18.0
 ```
 
 
 ### Language version numbers
 
-Dart language versions are identified by a major and minor number
-that match the first two components of the Dart SDK.
-For example, the latest language version supported by
-the 2.7.3 Dart SDK is Dart 2.7.
-Each Dart SDK supports all of the language versions covered
-by its major version number.
-That means that the 2.7.3 Dart SDK supports language
-versions 2.7, 2.6, 2.5, and so on, down to 2.0.
+Dart language versions follow semantic versioning with
+a major version number and a minor version number.
+Dart language versions can include a patch number.
+Patches should not change the language except for bug fixes.
+To illustrate: Dart 2.18.3 serves as the latest release of the
+Dart 2.18 SDK.
+Each Dart SDK supports all of the language versions within its major
+version number.
+That means that Dart SDK 2.18.3 supports language versions
+2.0 through 2.18 inclusive, but not Dart 1.x.
 
 Deriving the language version from the SDK version
 implies the following:
 
-* Whenever a minor version of the SDK ships,
-  a new language version appears.
-  In practice, many of these language versions are very similar to
-  and entirely compatible with previous versions.
-  For example, the Dart 2.9 language is essentially identical to
-  the Dart 2.8 language.
+* Whenever a minor version of the SDK ships, a new language version appears.
+  In practice, many of these language versions function in a very
+  similar manner to previous versions with full compatibility.
+  For example, the Dart 2.9 language works much like the Dart 2.8
+  language.
```

Comment:
It seems the changes lost the connection to [my comment](https://github.com/dart-lang/site-www/pull/4684#discussion_r1142087005) about this section, but my thoughts still stand:

> I think in general, I prefer the old text here. I think this change makes it say that the "language versions function in a similar manner when they have full compatibility", but the old text is saying that "many language versions are entirely compatible with each other".